packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/internal/LogObfuscator.kt

```diff
@@ -0,0 +1,248 @@
+        ).let {
+            GenericRegexPatternReplacer(it)
+        }
+
+        // Patterns used when sending a login with a JWT request:
+        // `"token":"<TOKEN>"` becomes `"token":"***"`
+        private fun loginJwt(): LogReplacer = mapOf(
+            """(("$TOKEN_KEY"):(\s?".+?"))""".toRegex() to """"$TOKEN_KEY":"***""""
+        ).let {
+            GenericRegexPatternReplacer(it)
+        }
+
+        // Creates a replacer that hides parameters for custom functions
+        private fun customFunction(): LogReplacer = CustomFunctionPatternReplacer
+    }
+}
+
+// Replacer for any given feature but custom functions
+// FIXME Access and refresh tokens cannot be replaced by regex due to
+//  https://github.com/realm/realm-kotlin/issues/1284
+//  so until we figure out what is wrong, do manual string replacement and hopefully nothing will
+//  change - if the message string changes our tests will hopefully catch it
+internal class GenericRegexPatternReplacer(
+    private val patternReplacementMap: Map<Regex, String>
+) : LogReplacer {
+    override fun findAndReplace(input: String): String {
+        return if (
+            input.contains("RESPONSE: 200 OK") &&
+            input.contains("access_token") &&
+            input.contains("refresh_token")
+        ) {
+            val beforeAccessToken = input.substringBefore(""""access_token"""")
+            val afterRefreshToken = input.substringAfter(""""user_id":""")
+            """$beforeAccessToken"access_token":"***","refresh_token":"***","user_id":$afterRefreshToken"""
+        } else {
+            var obfuscatedString = input
+            val entries: Set<Map.Entry<Regex, String>> = patternReplacementMap.entries
+            for (entry in entries) {
+                val pattern = entry.key
+                obfuscatedString = pattern.replace(obfuscatedString, entry.value)
+            }
+            obfuscatedString
+        }
+    }
+}
+
+// Replacer for custom function arguments. It combs the request and hides both the parameters that
+// are send and any possible received result
+internal object CustomFunctionPatternReplacer : LogReplacer {
+    override fun findAndReplace(input: String): String {
+        val (pattern, replacement) = when {
+            input.contains("REQUEST: ") ->
+                """("arguments"):\[.*]""".toRegex() to """"arguments":[***]"""
+            input.contains("RESPONSE: 200 OK") ->
+                """BODY START\n.*\nBODY END""".toRegex() to "BODY START\n***\nBODY END"
+            else -> return input
+        }
+        return pattern.replace(input, replacement)
+    }
+}
+
+internal object LogObfuscatorImpl : HttpLogObfuscator {
+
+    private val urlRegex =
+        Regex("https?:\\/\\/(www\\.)?[-a-zA-Z0-9@:%._\\+~#=]{1,256}\\.[a-zA-Z0-9()]{1,6}\\b([-a-zA-Z0-9()!@:%_\\+.~#?&\\/\\/=]*)")
+
+    private val features: Collection<ObfuscatorFeature> = setOf(
```

Comment:
Isn't this just the keys of `regexReplacerMap`? Why don't you just use something like
```
urlRegex.find(input)?.let { matchResult ->
   regexReplacerMap.filterKeys { url.container(it) }.values.map { it.findAndReplace (...) }
```